tests/v2/actions/observability/stackstate.go

```diff
@@ -0,0 +1,212 @@
+package observability
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/pkg/wait"
+	log "github.com/sirupsen/logrus"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	StackstateName         = "stackstate"
+	ObservabilitySteveType = "configurations.observability.rancher.io"
+	CrdGroup               = "observability.rancher.io"
+	ApiExtenisonsCRD       = "apiextensions.k8s.io.customresourcedefinition"
+)
+
+// NewStackstateConfiguration is a constructor that takes in the configuration and creates an unstructured type to install the CRD
+func NewStackstateCRDConfiguration(namespace string, stackstateCRDConfig StackStateConfigs) *unstructured.Unstructured {
+
+	crdConfig := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name":      StackstateName,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"url":          stackstateCRDConfig.Url,
+				"serviceToken": stackstateCRDConfig.ServiceToken,
+			},
+		},
+	}
+	return crdConfig
+}
+
+// InstallNodeDriver is a helper that utilizes the rancher client and add the stackstate domains to whitelist them.
+func InstallNodeDriver(client *rancher.Client, whitelistDomains []string) error {
```

Comment:
node driver is to use stackstate to create clusters for you, right? Or is this something else?

Reply:
this is to whitelist the stackstate domains

Reply:
will update the name to WhitelistStackstateDomain and add a comment that it gets resolved in future.